src/blueprints/Do.Blueprints.Service.Application/ServiceSpecExtensions.cs

```diff
@@ -218,28 +218,32 @@ public static void ShouldHaveOneParameter<T>(this MethodInfo source)
 
     #endregion
 
-    #region System
+    #region TimeProvider
 
-    public static ISystem TheSystem(this Mocker mockMe,
+    public static TimeProvider TheTime(this Mocker mockMe,
         DateTime? now = default,
-        bool passSomeTime = false
+        bool passSomeTime = false,
+        bool reset = false
     )
     {
-        var system = mockMe.Spec.GiveMe.The<ISystem>();
-        var mock = Mock.Get(system);
+        var fakeTimeProvider = (ResettableFakeTimeProvider)mockMe.Spec.GiveMe.The<TimeProvider>();
+
+        if (reset)
+        {
+            fakeTimeProvider.Reset();
+        }
 
         if (now is not null)
         {
-            mock.Setup(c => c.Now).Returns(now.Value);
+            fakeTimeProvider.SetUtcNow(new DateTimeOffset(now.Value));

```

Comment:
does this work correctly with the use of `.GetNow()` in business objects?

Reply:
`GetNow()` we wrote for the real `TimeProvider` and the `GetLocalNow` of the `FakeTimeProvider` will not behave the same.

Is the problem here that we give Local time but we assign it with `SetUtc`? Then I can write a helper like `SetLocalTime`. To be more consistent.

Reply:
yes the potential problem is that we set directly using utc without any offset, and business will use `GetNow` which is just wrapper around `GetLocalNow`. 

so we need a test case where;

1. we mock the time using this method
2. we insert an entity which sets its date time using `GetNow`
3. querying it from db using that datetime to assert if it returns inserted

i'm curious if it would cause any problem with local time zone set to something other than `0`